src/aspire/source/__init__.py

```diff
@@ -589,12 +590,14 @@ def save_metadata(
             axis=1,
         )
 
-        if save_images:
-            # Create a new column that we will be populating in the loop below
-            # For
-            df["_rlnImageName"] = ""
+        mrcs_filenames = None
+
+        with open(starfile_filepath, "w") as f:
+            if save_images:
+                # Create a new column that we will be populating in the loop below
+                # For
+                df["_rlnImageName"] = ""
```

Comment:
In line with the immutability of the `ImageSource`, we shouldn't be modifying the `df` here, right?

Reply:
This `df` is local variable copied from `self._metadata`. We did not modify the ImageSource object. This a general way we can add or modify some columns for output to a new star file.

Reply:
Ok.